src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -93,6 +155,7 @@ protected override void Dispose(bool disposing)
             state.CancellationToken = cancellationToken;
             state.RequestMessage = request;
             state.Handler = this;
+            state.ProxyPolicy = _proxyPolicy;
```

Comment:
the complete code getter and setter based on the update to the pull request is as follows:

        internal bool UseProxy
        {
            get
            {
                return _proxyPolicy != CurlProxyUsePolicy.DoNotUseProxy;
            }

            set
            {
                CheckDisposedOrStarted();
                if (value)
                {
                    _proxyPolicy = CurlProxyUsePolicy.UseCustomProxy;
                }
                else
                {
                    _proxyPolicy = CurlProxyUsePolicy.DoNotUseProxy;
                }
            }
        }